src/common/hooks/useStateSafely.ts

```diff
@@ -0,0 +1,21 @@
+import type { Dispatch, SetStateAction } from 'react';
+import { useCallback, useEffect, useRef, useState } from 'react';
```

Comment:
why do we have separate imports for functions and types ?

Reply:
A: in order to make it clear what types the module is consuming and which functions it uses from the React package.

Reply:
IMHO, it would make sense when applied consistently throughout the code, which is not recent state.

Reply:
IMHO, it's a good chance to start adopting it.

Reply:
I disagree with adopting this, unless we have automatic prettier/eslint rule for it

Reply:
Can anybody mention why this is a bad practice? So far this discussion is only a matter of "taste".
- Line 1 gets completely erased after the code is transpiled.
- It is not harming the project in any form.
- It helps readers differentiate between types and functions exported by the package.

*Honestly, this is not mandatory. I don't expect you to start following this convention if you don't like it.*

Anyway, [this rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/consistent-type-imports.md) should help (despite I don't think it should be using it).